IdentityCore/src/telemetry/MSIDTelemetryCacheEvent.m

```diff
@@ -108,13 +109,36 @@ - (void)setSpeInfo:(NSString *)speInfo
 
 - (void)setToken:(MSIDBaseToken *)token
 {
+    if (!token)
+    {
+        return;
+    }
+    
     [self setTokenType:token.tokenType];
     [self setSpeInfo:token.additionaServerlInfo[MSID_TELEMETRY_KEY_SPE_INFO]];
     
+    if (token.tokenType == MSIDTokenTypeLegacySingleResourceToken)
+    {
+        [self setIsRT:MSID_TELEMETRY_VALUE_YES];
+        [self setRTStatus:MSID_TELEMETRY_VALUE_TRIED];
+    }
+    
     if (token.tokenType == MSIDTokenTypeRefreshToken)
     {
         MSIDRefreshToken *refreshToken = (MSIDRefreshToken *)token;
-        [self setIsFRT:[NSString msidIsStringNilOrBlank:refreshToken.familyId] ? MSID_TELEMETRY_VALUE_NO : MSID_TELEMETRY_VALUE_YES];
+        
+        BOOL isFRT = [token.clientId isEqualToString:[MSIDTokenCacheKey familyClientId:refreshToken.familyId]];
```

Comment:
Just curious, is  `BOOL isFRT=refreshToken.familyId` good enough? :)
I notice that `[MSIDTokenCacheKey familyClientId:]` could return `foci-1` even if we give it a null value.

Reply:
Oh, seems the old code has a bug? Both MRRT and FRT could have familyId being something. Then `[token.clientId isEqualToString:[MSIDTokenCacheKey familyClientId:refreshToken.familyId]` is the only way to know if it is a FRT. Am I correct? :)

Reply:
Yes, we want to know if the current entry is FRT, not if the refresh token was issued for a family. Both MRRT and FRT will have the familyId flag.